### PR TITLE
Fix live-updates for amenities

### PR DESCRIPTION
**Pull request by ivanPyrohivskyi**
To issue https://github.com/osmandapp/OsmAnd/issues/17898

Best place for test in Berlin is here (added opening hours):
https://www.openstreetmap.org/node/886088999

### CODE REVIEW

File: OsmAnd/src/net/osmand/plus/resources/ResourceManager.java [hunk 1]
```diff
@@ -1150,10 +1151,10 @@ public List<AmenityIndexRepository> getAmenityRepositories() {
 		return getAmenityRepositories(true);
 	}
 
-	public List<AmenityIndexRepository> getAmenityRepositories(boolean includeTravel) {
-		List<String> fileNames = new ArrayList<>(amenityRepositories.keySet());
+	public LinkedList<AmenityIndexRepository> getAmenityRepositories(boolean includeTravel) {
+		LinkedList<String> fileNames = new LinkedList<>(amenityRepositories.keySet());
 		Collections.sort(fileNames, Algorithms.getStringVersionComparator());
-		List<AmenityIndexRepository> res = new ArrayList<>();
+		LinkedList<AmenityIndexRepository> res = new LinkedList<>();
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1157]: ? Linked list worse by performance than ArrayList, don't see any reason below

File: OsmAnd/src/net/osmand/plus/resources/ResourceManager.java [hunk 2]
```diff
@@ -1194,16 +1196,31 @@ public List<Amenity> searchAmenities(SearchPoiTypeFilter filter, double topLatit
...
 		} finally {
 			searchAmenitiesInProgress = false;
 		}
-		return amenities;
+
+		// filter amenities for live-updates map
+		HashSet<Long> amenitiesHash = new HashSet<>();
+		List<Amenity> res = new ArrayList<>();
+		for (Map.Entry<Integer, List<Amenity>> entry : sortedResult.entrySet()) {
+			for (Amenity a : entry.getValue()) {
```
Comment (vshcherb) [inline OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1212]: >> sortedResult.entrySet()
Is nto sorted at all what the reason?

How is it different than old approach

Comment (ivanPyrohivskyi) [reply OsmAnd/src/net/osmand/plus/resources/ResourceManager.java:1212]: List<Amenity> r = index.searchAmenities(...) - synchronize method
amenities.addAll(r); - at the adding moment r is empty, so adding only link of r, but not r values(), values are adding randomly (depend from speed of *.obf reading).
Complete list amenities available only _try { }_ block
So I strictly set order by reading indexes.

**Review by vshcherb — CHANGES_REQUESTED** [review]
Code changes needed

**Pull request closed without merging**